metricflow/query/query_parser.py

```diff
@@ -318,7 +318,12 @@ def _parse_and_validate_query(
                 f"{time_constraint_start.isoformat()}"
             )
 
-        time_constraint = TimeRangeConstraint(start_time=time_constraint_start, end_time=time_constraint_end)
+        time_constraint: Optional[TimeRangeConstraint] = TimeRangeConstraint(
+            start_time=time_constraint_start, end_time=time_constraint_end
+        )
+        if time_constraint == TimeRangeConstraint.all_time():
```

Comment:
Might be easier to understand if you update the series of conditionals above.

Reply:
I think we can simplify the above conditionals by creating a `TimeRangeConstraint` with whatever is provided then just do an intersection with ALL TIME, but we would lose the warnings messages.